src/cmdlib.sh

```diff
@@ -429,7 +429,8 @@ runvm() {
 
     # tmp_builddir is set in prepare_build, but some stages may not
     # know that it exists.
-    export tmp_builddir="${tmp_builddir:-${workdir}/tmp/build${IMAGE_TYPE:+.$IMAGE_TYPE}}"
+    # shellcheck disable=SC2086
+    export tmp_builddir="${tmp_builddir:-$(mktemp -p ${workdir}/tmp -d supermin.XXXX)}"
```

Comment:
Will this leave a bunch of `supermin.XXXX` directories around? I don't think the root of `$tmp_builddir` ever gets cleaned up.

Reply:
No
https://github.com/coreos/coreos-assembler/blob/master/src/cmd-buildextend-metal#L301
https://github.com/coreos/coreos-assembler/pull/1690/files#diff-496abb20f29d0275d6697b14356e60e1R531

The only way it would leave them if a failure happens during the `runvm` stage. We probably need some hygiene around `${workdir}/tmp`, but that's out of scope for this PR.

Reply:
I just did a build and I see a few getting left around:

```
$ cosa fetch --strict && cosa build --strict && cosa buildextend-metal && cosa buildextend-metal4k && cosa buildextend-live

$ ls -ld tmp/supermin.* 
drwxr-xr-x. 2 dustymabe dustymabe 4096 Aug 28 15:05 tmp/supermin.build
drwx------. 2 dustymabe dustymabe 4096 Sep  1 13:31 tmp/supermin.ENO6
drwxr-xr-x. 2 dustymabe dustymabe 4096 Aug 28 15:05 tmp/supermin.prepare
drwx------. 2 dustymabe dustymabe 4096 Sep  1 13:33 tmp/supermin.r64z
```

If I `grep -r runvm` I see it in a few more places than just the metal case.

If we don't think it's worth cleaning them up we can merge this to get us unblocked.

Reply:
note that the `supermin.build` and `supermin.prepare` dirs are from Aug 28 so they can probably be ignored.